src/components/iconSwitch/Thumb.tsx

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import { Animated, StyleSheet, View } from "react-native";
+import Star from "../../assets/icons/Star";
+
+type ThumbProps = {
+  range: number[];
+  animatedValue: Animated.Value;
+  width: number;
+  height: number;
+  colors: {
+    true: string;
+    false: string;
+  };
+  size: number;
+  value: boolean;
+  iconColors: {
+    true: string;
+    false: string;
+  };
+  pressIndicator: boolean;
+};
+
+export default function Thumb(props: ThumbProps) {
+  const { range, animatedValue, width, size, value, colors } = props;
+
+  const radius = Math.round(size / 2);
+
+  const color = animatedValue.interpolate({
+    inputRange: range,
+    outputRange: [colors.false, colors.true],
+  });
+
+  const position = animatedValue.interpolate({
+    inputRange: range,
+    outputRange: [0, width - size],
+  });
+
+  const pressedIndicatorPosition = animatedValue.interpolate({
+    inputRange: range,
+    outputRange: [0 - radius, width - radius - size],
+  });
+
+  return (
+    <>
+      {props.pressIndicator && (
+        <Animated.View
+          style={[
+            styles.pressedIndicator,
+            {
+              backgroundColor: color,
+              left: pressedIndicatorPosition,
+              width: size * 2,
+              borderRadius: size,
+              top: 0 - radius,
+              aspectRatio: 1,
+            },
```

Comment:
Uhm can these styles be defined separately?

Reply:
color, pressedIndicatorPosition, size, and radius are either calculated or passed as props. May be hard to move them.